dotnet-roslyn/analyzers/Audacia.CodeAnalysis.Analyzers/src/Audacia.CodeAnalysis.Analyzers/Rules/NestedControlStatements/NestedControlStatementsAnalyzer.cs

```diff
@@ -130,10 +130,35 @@ private static void AnalyzeStatement(SyntaxNodeAnalysisContext context, SyntaxNo
 
         private static void ReportAtContainingSymbol(int depth, int maxDepth, SyntaxNodeAnalysisContext context, SyntaxNode node)
         {
-            string kind = node.Kind().ToString();
+            var baseKind = node.Kind();
+            var kind = baseKind.ToString();
             var location = node.GetLocation();
 
             context.ReportDiagnostic(Diagnostic.Create(Rule, location, kind, depth, maxDepth));
+
+            // Else and If Else Clauses are considered to be a property of an IfStatement, so we need to handle them separately.
+            if (baseKind is SyntaxKind.IfStatement)
+            {
+                ReportElseStatement(baseKind, context, node, depth, maxDepth);
+            }
+        }
+        
+        private static void ReportElseStatement(SyntaxKind kind, SyntaxNodeAnalysisContext context, SyntaxNode node, int depth, int maxDepth)
```

Comment:
My understanding then, is the existing code was detecting the too much nesting within the else block, it just wasn't being reported on? So you have just had to update the reporting logic so it is shown to the user?